src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -16,21 +16,21 @@ end
 
 #Constructors
 function Cone(;rtopMin = 0, rtopMax = 1, rbotMin = 0, rbotMax = 1, φMin = 0, φMax = 2π, zMin = -1/2, zMax = 1/2)
-    T = promote_type(typeof.((rtopMin, rtopMax, rbotMin, rbotMax, φMin, φMax, zMin, zMax))...)
+    T = float(promote_type(typeof.((rtopMin, rtopMax, rbotMin, rbotMax, φMin, φMax, zMin, zMax))...))
     rtop, rbot = if rtopMin == rbotMin == 0 
         T(rtopMax), T(rbotMax)
     else
         T(rtopMin)..T(rtopMax), T(rbotMin)..T(rbotMax)
     end
-    φ = mod(φMax - φMin, 2π) == 0 ? nothing : T(φMin)..T(φMax)
+    φ = mod(T(φMax) - T(φMin), T(2π)) == 0 ? nothing : T(φMin)..T(φMax)
     z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
     rtop == rbot ? Tube( T, rtop, φ, z) : Cone( T, rtop, rbot, φ, z)
 end
 Cone(rtopMin, rtopMax, rbotMin, rbotMax, φMin, φMax, zMin, zMax) = Cone(;rtopMin, rtopMax, rbotMin, rbotMax, φMin, φMax, zMin, zMax)
 
-function Cone(rtop::R1, rbot::R2, height::H) where {R1, R2, H}
-    T = promote_type(R1, R2, H)
-    Cone( T, T(rtop), T(rbot), nothing, T(height/2))
+function Cone(rtop::R1, rbot::R2, height::H) where {R1<:Real, R2<:Real, H<:Real}
```

Comment:
Lets restrict it to `AbstractFloat`. `Real` includes e.g. `Int`. 
And if we do restrict the type in general we should do it in the base construct of your primitives.
If we want to support units in the future we can still change/adapt to it. 
For now I would just add constructors which convert units into SI units and strip the units away.

Reply:
Everywhere? Then also in the functions that are defined in the struct?